lightning/src/ln/functional_tests.rs

```diff
@@ -7298,3 +7298,22 @@ fn test_bump_txn_sanitize_tracking_maps() {
 		}
 	}
 }
+
+#[test]
+fn test_override_channel_config() {
+    let chanmon_cfgs = create_chanmon_cfgs(2);
+    let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+    let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+    let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+    // Node0 initiates a channel to node1 using the override config.
+    let mut override_config = UserConfig::default();
+    override_config.own_channel_config.our_to_self_delay = 200;
```

Comment:
It looks like the existing testing for bad UserConfig values is pretty light -- this test could be a good opportunity to add some more :)